core/src/main/scala/kafka/controller/TopicDeletionManager.scala

```diff
@@ -292,22 +291,35 @@ class TopicDeletionManager(config: KafkaConfig,
    * @param replicasForTopicsToBeDeleted
    */
   private def startReplicaDeletion(replicasForTopicsToBeDeleted: Set[PartitionAndReplica]): Unit = {
-    replicasForTopicsToBeDeleted.groupBy(_.topic).keys.foreach { topic =>
-      val aliveReplicasForTopic = controllerContext.allLiveReplicas().filter(p => p.topic == topic)
-      val deadReplicasForTopic = replicasForTopicsToBeDeleted -- aliveReplicasForTopic
+    val allDeadReplicasForTopic = mutable.Set.empty[PartitionAndReplica]
```

Comment:
Not sure it matters too much, but because of the way these are collected, I don't think they need to be sets. We could use `ListBuffer` instead, which is already a `Seq`.

Reply:
You are absolutely right. I thought that a `Set` was required because we add `Set`s but `ListBuffer` works perfectly.